components/TodoList.js

```diff
@@ -0,0 +1,77 @@
+import renderTodoListTemplate from "../utils/renderTodoListTemplate";
+
+export default function TodoList({
+  $target,
+  initialState,
+  onDeleteTodo,
+  onToggle,
+  onChangeMode,
+  onEdit,
+}) {
+  this.state = initialState;
+
+  const $todoList = document.createElement("ul");
+  $todoList.className = "todo-list";
+  $target.appendChild($todoList);
+
+  this.setState = (nexState) => {
+    this.state = nexState;
+    this.render();
+  };
+
+  this.render = () => {
+    const todoListTemplate = renderTodoListTemplate(this.state);
+    $todoList.innerHTML = todoListTemplate;
+  };
+
+  this.render();
+
+  // Todo 요소 클릭 (삭제 OR 토글)
+  const handleClick = ({ target }) => {
+    const $todo = target.closest("li");
+
+    if ($todo) {
+      const { id } = $todo.dataset;
+      const { className } = target;
+
+      if (className === "destroy") {
+        onDeleteTodo(id);
+      } else if (className === "toggle") {
+        onToggle(id);
+      }
+    }
```

Comment:
`early return` 을 사용해서 코드의 depth를 줄여보셔도 좋을거 같아요~
```suggestion
    if(!$todo) return;

    const { id } = $todo.dataset;
    const { className } = target;

    if (className === "destroy") {
      onDeleteTodo(id);
    } else if (className === "toggle") {
      onToggle(id);
    }
```